src/Native/Unix/System.Security.Cryptography.Native/opensslshim.h

```diff
@@ -1036,16 +1036,14 @@ FOR_ALL_OPENSSL_FUNCTIONS
 #define X509_get_X509_PUBKEY local_X509_get_X509_PUBKEY
 #define X509_get_version local_X509_get_version
 #define X509_up_ref local_X509_up_ref
-
-#if OPENSSL_VERSION_NUMBER < OPENSSL_VERSION_1_0_2_RTM
```

Comment:
The functions require a version test, there’s no ifndef equivalent.

Your change suggests the no partial wildcards define was removed in a patch release, which seems unlikely. Can you confirm that is the actual problem?

Reply:
The actual problem is a build failure with following error
```
  [ 82%] Linking C shared library System.Security.Cryptography.Native.OpenSsl.dylib
  Undefined symbols for architecture x86_64:
    "_X509_STORE_CTX_get0_store", referenced from:
        _CryptoNative_X509StoreCtxReset in pal_x509.c.o
        _CryptoNative_X509StoreCtxResetForSignatureError in pal_x509.c.o
        _CheckOcsp in pal_x509.c.o
  Failed to build corefx native components.
       (maybe you meant: _local_X509_STORE_CTX_get0_store)
    "_X509_check_host", referenced from:
        _CryptoNative_CheckX509Hostname in openssl.c.o
       (maybe you meant: _local_X509_check_host)
  ld: symbol(s) not found for architecture x86_64
```

Reply:
The build problem is quite reproducible on macOS 10.14 + OpenSSL from Homebrew. I keep running into it too when I forget to use OpenSSL 1.1.x.

Reply:
https://github.com/openssl/openssl/blob/OpenSSL_1_0_2-stable/util/libeay.num#L4346 says that function exists in all builds of OpenSSL 1.0.2 (no conditional compilation).

Before taking any change here I'd like to understand why it's missing in homebrew's build (if it's something they went out of their way to remove then we'd need to improve detection for their non-conforming library)

Reply:
I just tried on my mac, and I don't get this error.  (Mojave, brew-installed 1.0.2r)  https://github.com/dotnet/corefx/issues/38445#issuecomment-500888233

Reply:
Did you build with clean git and `OPENSSL_ROOT_DIR=/usr/local/Cellar/openssl/1.0.2r ./build.sh` ?

Reply:
I built with git clean, using just build-native.sh; and registered pkgconfig instead of setting OPENSSL_ROOT_DIR (https://github.com/dotnet/corefx/wiki/Development-environment-setup-for-MacOS#openssl)

Reply:
I used `OPENSSL_ROOT_DIR` just like Marek. Just running `./build.sh` actually suggests to do that which is probably why we used that way.

Reply:
Based on the linker output in #38445, it looks like setting OPENSSL_ROOT_DIR fixed finding the headers, but the linker was picking /usr/lib; which means that builds using the environment variable were actually linking to (and presumably running-with) the out-of-support 0.9.8... unless this is just that something recently changed in the openssl-related files feeding into cmake.